language_service/src/completion.rs

```diff
@@ -156,27 +205,27 @@ impl CompletionListBuilder {
 
         let display = CodeDisplay { compilation };
 
-        self.push_sorted_completions(
-            Self::get_callables(current, &display),
-            CompletionItemKind::Function,
-        );
-        self.push_sorted_completions(
-            Self::get_callables(std, &display),
-            CompletionItemKind::Function,
-        );
-        self.push_sorted_completions(
-            Self::get_callables(core, &display),
-            CompletionItemKind::Function,
-        );
-        self.push_completions(Self::get_namespaces(current), CompletionItemKind::Module);
-        self.push_completions(Self::get_namespaces(std), CompletionItemKind::Module);
-        self.push_completions(Self::get_namespaces(core), CompletionItemKind::Module);
+        self.push_sorted_completions(Self::get_callables(current, &display));
+        self.push_sorted_completions(Self::get_std_callables(
```

Comment:
I see you only special-case the standard library. What if I declare two namespaces in the same file and reference one from the other? Is there a way to make the function more generic so that it's applicable to all `Package`s in the compilation, including the current one?

This would also set us up nicely for the future when we support multiple files.

Reply:
I've made the change to support auto-opens for callables in other namespaces. While doing so I discovered that it is possible in Q# to have nested operations in Q#, which are difficult to support in the current implementation of completions. I've had to disallow these in this PR to avoid completion items that users don't actually have access to. We'll have to work on supporting this in the future.